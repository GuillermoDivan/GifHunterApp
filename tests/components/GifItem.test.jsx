import { render, screen } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';

describe('Tests on <GifItem />', () => {

    const title = 'Pedro Pascal';
    const url = 'http://pedropascal.com/pedropascal123.jpg'

    test('should match snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect (container).toMatchSnapshot();
    });

    test('should show image with the proper url and alt', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title); 
    });   
    
    test('should show title on the component', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });  

});