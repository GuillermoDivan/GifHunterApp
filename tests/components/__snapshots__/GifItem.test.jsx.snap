// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pedro Pascal"
      src="http://pedropascal.com/pedropascal123.jpg"
    />
    <p>
      Pedro Pascal
    </p>
  </div>
</div>
`;
